diff --git a/scripts/dtc/dtc-lexer.lex.c b/scripts/dtc/dtc-lexer.lex.c
index 6ac551375..a09873adc 100644
--- a/scripts/dtc/dtc-lexer.lex.c
+++ b/scripts/dtc/dtc-lexer.lex.c
@@ -611,7 +611,7 @@ char *yytext;
 #include "srcpos.h"
 #include "dtc-parser.tab.h"
 
-YYLTYPE yylloc;
+extern YYLTYPE yylloc;
 extern bool treesource_error;
 
 /* CAUTION: this will stop working if we ever use yyless() or yyunput() */
